core/src/test/scala/org/genivi/sota/core/UpdateServiceSpec.scala

```diff
@@ -132,6 +128,53 @@ class UpdateServiceSpec extends PropSpec
     }
   }
 
+  property("queuing an update for a blacklisted package fails") {
+    val newPackage = PackageGen.sample.get
+    val req = updateRequestGen(defaultNs, PackageIdGen).sample.get.copy(packageId = newPackage.id)
+
+    val f = for {
+      packageM <- db.run(Packages.create(newPackage))
+      _ <- BlacklistedPackages.create(packageM.namespace, packageM.id)
+      _ <- service.queueUpdate(req, _ => Future.successful(Map.empty))
+    } yield packageM
+
+    val e = f.failed.futureValue
+
+    e shouldBe Errors.BlacklistedPackage
+  }
+
+  property("queuing a device update for a blacklisted package fails") {
+    val device = genDevice.sample.get
+    val newPackage = PackageGen.sample.get
+
+    val f = for {
+      packageM <- db.run(Packages.create(newPackage))
+      _ <- BlacklistedPackages.create(packageM.namespace, packageM.id)
+      _ <- service.queueDeviceUpdate(device.namespace, device.id, packageM.id)
+    } yield packageM
+
+    f.failed.futureValue shouldBe Errors.BlacklistedPackage
```

Comment:
because `e` is for a `Future[Throwable]`, while `f` is for a `Future[Package]`. But I see the problem, I will use better val names.